packages/astro/src/core/build/plugins/plugin-manifest.ts

```diff
@@ -234,6 +236,32 @@ function buildManifest(
 			styles,
 			routeData: serializeRouteData(route, settings.config.trailingSlash),
 		});
+
+		/**
+		 * logic meant for i18n domain support, where we fill the lookup table
+		 */
+		const i18n = settings.config.experimental.i18n;
+		if (i18n && i18n.domains && i18n.routingStrategy === 'domain') {
+			// Domain routing strategy assumes that the locale starts at the beginning of the path, so it's safe to get the first item
+			// This is something like /es/guides/getting-started
+			const segments = route.route.split('/');
+			const pathnameLocale = segments[1];
+			const maybeLocaleDomain = i18n.domains[pathnameLocale];
+			if (maybeLocaleDomain) {
+				// This is something like guides/getting-started
+				let pathnameWithoutLocale = segments
+					.filter((segment) => segment !== pathnameLocale)
+					.slice(1)
+					.join('/');
+				if (shouldAppendForwardSlash(settings.config.trailingSlash, settings.config.build.format)) {
+					pathnameWithoutLocale = appendForwardSlash(pathnameWithoutLocale);
+				}
+				// We build a URL object like this: https://es.example.com/guides/getting-started
+				const domainPath = new URL(pathnameWithoutLocale, maybeLocaleDomain);
+				// We map  https://es.example.com/guides/getting-started -> /es/guides/getting-started
+				domainLookupTable[domainPath.toString()] = route.route;
```

Comment:
Hm, I don't think this will work for dynamic routes, right? In SSR essentially any route is possible, as long as there is a dynamic route that matches.

I was thinking this table would be a mapping of just the domain name part, to the associated locale. So for example:

```js
const domainLookupTable = {
	"https://fr.example.com": "fr",
	"https://en.example.com": "en-US",
}
```

I'm including the protocol because I think you said that it needs to be included.

So then during SSR it uses this lookup to see which locale the domain is for. And once it has the locale it uses that to call `matchRoute` will the locale as part of the pathname to get the correct route.

Does this make sense?

Reply:
This would make the logic much simpler. Sounds good!